airbyte-webapp/src/hooks/services/useConnectionHook.tsx

```diff
@@ -74,20 +79,43 @@ export function useConnectionService() {
 
 export const useSyncConnection = () => {
   const service = useConnectionService();
+  const webConnectionService = useWebConnectionService();
+  const { trackError } = useAppMonitoringService();
+  const queryClient = useQueryClient();
   const analyticsService = useAnalyticsService();
+  const notificationService = useNotificationService();
+  const workspaceId = useCurrentWorkspace();
+  const { formatMessage } = useIntl();
 
-  return useMutation((connection: WebBackendConnectionRead | WebBackendConnectionListItem) => {
-    analyticsService.track(Namespace.CONNECTION, Action.SYNC, {
-      actionDescription: "Manual triggered sync",
-      connector_source: connection.source?.sourceName,
-      connector_source_definition_id: connection.source?.sourceDefinitionId,
-      connector_destination: connection.destination?.destinationName,
-      connector_destination_definition_id: connection.destination?.destinationDefinitionId,
-      frequency: getFrequencyFromScheduleData(connection.scheduleData),
-    });
-
-    return service.sync(connection.connectionId);
-  });
+  return useMutation(
+    (connection: WebBackendConnectionRead | WebBackendConnectionListItem) => {
+      analyticsService.track(Namespace.CONNECTION, Action.SYNC, {
+        actionDescription: "Manual triggered sync",
+        connector_source: connection.source?.sourceName,
+        connector_source_definition_id: connection.source?.sourceDefinitionId,
+        connector_destination: connection.destination?.destinationName,
+        connector_destination_definition_id: connection.destination?.destinationDefinitionId,
+        frequency: getFrequencyFromScheduleData(connection.scheduleData),
+      });
+
+      return service.sync(connection.connectionId);
+    },
+    {
+      onError: (error: Error) => {
+        trackError(error);
+        notificationService.registerNotification({
+          id: `tables.startSyncError.${error.message}`,
+          text: `${formatMessage({ id: "connection.startSyncError" })}: ${error.message}`,
+          type: ToastType.ERROR,
+        });
+      },
+      onSuccess: async () => {
+        webConnectionService
```

Comment:
We may need to `await` here. I noticed that the button goes into a non-loading state for the time the list is loading.

Reply:
good point! fixed ✅